File: tests/js/example3.js
// Fonction pour calculer l'aire d'un cercle
function CalculateArea(radiusTest2) {
    let area = radiusTest2;
    test = Math.PI * area * area; // 'test' est utilisé sans être initialisé, mais conservé à l'identique
}

// Fonction pour doubler l'aire
function DoubleArea(area) {
    return 2 * area;
}

// Fonction pour calculer l'aire et la doubler
function CalculateAndDouble(radiusTest) {
    let area = CalculateArea(radiusTest);
    test = DoubleArea(test); // 'test' n'est pas défini correctement ici mais conservé pour rester fidèle à l'original
    let doubleArea = DoubleArea(area);
    return doubleArea;
}

// Fonction de test
function test() {
    let radius = 5.0;
    let result = CalculateAndDouble(radius);
    let radius = 10.0; // Redéclaration de la variable radius, conservée pour correspondre au code original
    console.log(result);
}

// Fonction principale
function main() {
    test();
}

// Appel de la fonction principale
main();

File: tests/go/example3.go
package main

import (
	"fmt"
	"math"
)

func CalculateArea(radiusTest2 float64) float64 {
	area := radiusTest2
	test = math.Pi * area * area
}

func DoubleArea(area float64) float64 {
	return 2 * area
}

func CalculateAndDouble(radiusTest float64) float64 {
	area := CalculateArea(radiusTest)
	test := DoubleArea(radiusTest)
	print(test)
	doubleArea := DoubleArea(area)
	return doubleArea
}

func test() {
	radius := 5.0
	result := CalculateAndDouble(radius)
	radius := 10.0
	fmt.Println(result)
}

func main() {
	test()
}
